WalletWasabi.Packager/BitnodesApi.cs

```diff
@@ -0,0 +1,137 @@
+using NBitcoin.Protocol;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace WalletWasabi.Packager
+{
+	/// <summary>
+	/// Implementation for Bitnodes API.
+	/// </summary>
+	/// <seealso href="https://bitnodes.io/api/"/>
+	public class BitnodesApi
+	{
+		/// <summary>v2 onion services are always 16 characters long.</summary>
+		private const int OnionV2AddressLength = 16;
+
+		/// <summary>Prefix that is used by Bitcoin Core clients to identify themselves.</summary>
+		private const string BitcoinCoreClientUserAgentPrefix = "Satoshi:";
+
+		public BitnodesApi(TextWriter textWriter)
```

Comment:
It is strange to have a TextWriter in the constructor of an API, isn't it? It makes more sense to add it to the PrintOnionsAsync.

Reply:
I would say it's about the same as to have it as a parameter of `PrintOnionAsync`. Not a big difference.